### DISCUSSION TITLE
Lesson 1 Exercise 9..extracting specific sequence "AACCGGUU"

### DISCUSSION THREAD

**Discussion by GitTodojetaime** [sandbox-bio/sandbox.bio#108] [Q&A] [answered]
I am on lesson exercise 9 of the beginners seqkit tutorial. I am using this to extract sequences from hairpin.fa 

seqkit grep -f ids.text hairpins.fa

5 patterns become loaded from the file with this command. What changes can I make to extract sequences with "AACCGGUU" specifically?

**Comment by robertaboukhalil** [comment]
Yes using `seqkit grep -f` can work here but you need to make sure the file `ids.text` only contains the sequences you're interested in (i.e. only AACCGGUU).

Also, note that `seqkit grep` by default matches patterns in the sequence name, not the sequence itself. Check out the [manual](https://bioinf.shenwei.me/seqkit/usage/#grep) for which flag you need to add in order to search for the pattern by sequence.

**Reply by GitTodojetaime** [reply]
Thank you. :)

**Reply by GitTodojetaime** [reply]
On another note...I am trying to Use [seqkit seq] to create the file formatted.fa, which formats hairpins.fa to show 80 bases per line instead of the current 60 bases per line.I am using this code..

seqkit seq --max-len 80 hairpins.fa

It is generating an output that looks correct...sequences roughly 80 bp...but the tutorial I'm using isn't marking it as correct. I would love to know your thoughts.

**Reply by GitTodojetaime** [reply]
> Thank you. :)

So I used the manual and found Region  R- and I need to include the nucleotides as a "1 based index"...where A is 1, T is 2, G is 3, C is 4...but what about U? I notice there is an 'N" in the index..and I'm assuming this might stand for Uracil.

**Reply by GitTodojetaime** [reply]
Now I'm seeing -s might be for sequence..still not sure how to specify the sequence..

**Reply by robertaboukhalil** [reply, marked as answer]
`seqkit seq --max-len 80 hairpins.fa` would only show you sequences that have length 80, which is not what we need. We need sequences of all length, but only showing 80 nucleotides per line, which the `--line-width` does

Yes, you can use `-s` for sequence and then use `--pattern` to specifying the sequence pattern to search for

Also, just in case it's not clear in the UI, you need to redirect the output of the tool to the file specified in the exercise.

**Reply by GitTodojetaime** [reply]
Thank you for the clarification..I misread the challenge :) The file is called formatted.fa which I have already created..
This is the command I tried seqkit seq --line-width 80 formatted.fa
Not...working
Any recommendations on redirecting the file?

**Reply by robertaboukhalil** [reply]
On the command line, you need to use `command > output.txt` to redirect the output of `command` to `output.txt`, i.e. `seqkit seq -w 80 hairpins.fa > formatted.fa`

**Reply by Law0804** [reply]
I did try this: seqkit seq -w 80 hairpins.fa > formatted.fa and did search in stackoverflow, but still, it did not work.  Any help?

**Reply by robertaboukhalil** [reply]
@Law0804 Sorry, there's a bug in the exercise checking! I'll deploy a fix soon.

**Reply by robertaboukhalil** [reply]
@Law0804 Should be fixed now, thanks so much for bringing it to my attention!
